### DISCUSSION TITLE
Can’t call OrderedSet.empty

### DISCUSSION THREAD

**Discussion by mxcl** [rimbu-org/rimbu#117] [Q&A] [answered]
rimbu 0.12.3 via Deno:

I've tried a variety of imports, with the same error, current code is:

```ts
import { OrderedSet } from "rimbu/ordered/mod.ts"

const foo = OrderedSet.empty<string>()
```

which gives the error:

```
Property 'empty' does not exist on type 'OrderedSetCreators'.deno-ts(2339)
```

However, `empty()` does work with `OrderedSortedSet`, it’s just I don’t want the order to be sorted.

**Comment by vitoke** [comment, marked as answer]
The ordered collections take an existing collection and add the order administration to them. Therefore, the `OrderedSet` can be seen as an interface that can be used for all ordered sets.

To instantiate ordered sets, there are the following options:
- use the `OrderedHashSet` to have an ordered collection that internally uses a `HashSet`. So this would become `OrderedHashSet.empty<string>()`
- use the `OrderedSortedSet` to have an ordered collection that internally uses a `SortedSet`. So this would become `OrderedSortedSet.empty<string>()`
- Create a custom context with a configured collection: `OrderedSet.createContext(...options...).empty<string>()`

If you don't need your collection to be sorted, the `OrderedHashSet` would be the best and easiest option.

Let me know if you need more information.

**Reply by mxcl** [reply]
Thank you. I found a few references to `OrderedSet.empty()` in the docs, will submit a PR to correct this if I can find them again.

**Reply by vitoke** [reply]
I see. Thanks for sharing, I'll also do a search. I do see now that there are incorrect mentions of `OrderedSet.of(...)` in the README. I've created a [PR](https://github.com/rimbu-org/rimbu/issues/118) for it.
